apps/api/src/app/change/usecases/promote-layout-change/promote-layout-change.use-case.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@nestjs/common';
+import { LayoutEntity, LayoutRepository } from '@novu/dal';
+
+import { PromoteTypeChangeCommand } from '../promote-type-change.command';
+
+@Injectable()
+export class PromoteLayoutChange {
+  constructor(private layoutRepository: LayoutRepository) {}
+
+  async execute(command: PromoteTypeChangeCommand) {
+    const item = await this.layoutRepository.findOne({
+      _environmentId: command.environmentId,
+      _id: command.item._id,
+    });
+
+    const newItem = command.item as LayoutEntity;
+
+    if (!item) {
+      const layoutEntity = {
+        ...newItem,
+        _creatorId: command.userId,
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
```

Comment:
And here you will be creating a layout with _parentId being the _id of the item you are promoting.